alerting/src/main/kotlin/org/opensearch/alerting/DocumentReturningMonitorRunner.kt

```diff
@@ -410,7 +491,47 @@ object DocumentReturningMonitorRunner : MonitorRunner {
         return response.hits
     }
 
+    private fun getMatchedQueries(
+        monitorCtx: MonitorRunnerExecutionContext,
+        docs: List<BytesReference>,
+        monitor: Monitor
+    ): SearchHits {
+        val percolateQueryBuilder = PercolateQueryBuilderExt("query", docs, XContentType.JSON)
+
+        val searchRequest = SearchRequest(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)
+        val searchSourceBuilder = SearchSourceBuilder()
+        searchSourceBuilder.query(percolateQueryBuilder)
+        searchRequest.source(searchSourceBuilder)
+
+        return if (monitorCtx.clusterService!!.state().routingTable.hasIndex(ScheduledJob.DOC_LEVEL_QUERIES_INDEX)) {
+            val response: SearchResponse = monitorCtx.client!!.execute(SearchAction.INSTANCE, searchRequest).actionGet()
```

Comment:
passing `monitor.id` as a filter also so that it filters by monitor id & then matches only those queries. https://github.com/opensearch-project/alerting/blob/6328f06451ced797f66d481f47a9fd757944f116/alerting/src/main/kotlin/org/opensearch/alerting/DocumentReturningMonitorRunner.kt#L464-#L477